### ISSUE TITLE
GoToDefinition on method implementation should go to overridden method.

### ISSUE THREAD

**Issue by OmarTawfik** [dotnet/roslyn#18716]
```csharp
public abstract class Parent
{
    public abstract void Method();
}

public class Child : Parent
{
    public override void Method()
    {
    }
}
```

If I use `go to definition` or press `F12` on `Child.Method` name, I expect it to go to `Parent.Method` (the overridden method). I find this tiny feature to be very helpful instead of manually looking up the inheritance chain to find out which parent class or interface is defining the abstract or virtual parent.

The same is true for parent type (`class` or `interface`), and member type (`method`, `property`, etc...).

**Comment by Pilchie** [comment]
Related to https://github.com/dotnet/roslyn/issues/6544.  I'm leaning toward making both of these changes.

**Comment by jcouv** [comment]
I'm glad to hear that you ran into the same problem and naturally arrived at the same solution: GoToDefinition triggered on a definition should go "up" to overridden and implemented methods.

I prototyped this (https://github.com/dotnet/roslyn/pull/22963) and it felt pretty natural/usable.
The next step is for the IDE team to confirm this design fits well into broader navigation plans, then we can push ahead.

**Comment by CyrusNajmabadi** [comment]
> The next step is for the IDE team to confirm this design fits well into broader navigation plans, then we can push ahead.

FWIW, i think your approach seems quite reasonable and should be done.

**Comment by jcouv** [comment]
Fixed by #22963 which associates the "override" keyword with the overridden symbol (in the IDE).
